### PR TITLE
refactor(app): add robot is busy modal to choose robot slideout

### LINKED ISSUE #9775: Run Details Page: Clear a protocol run from the robot
# Overview
We need a way to clear protocols runs that are not-started/cancelled/failed/completed from the robot before the next protocol run can be started. This ticket will cover all of the changes required to add this functionality. 

# Design 
The flowcharts and final designs related to this work can be found here: https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=7269%3A267907

# Implementation Details
On the protocol run details page, a user should be shown the button to clear a protocol that's been loaded onto the robot in the following scenarios: 

- A protocol has been loaded but not started yet
- A protocol is currently running (the clear button protocol should be shown but disabled)
- A protocol run has been cancelled
- A protocol run has failed
- A protocol run has completed

In addition to the clear protocol button, a banner with a clear protocol link will appear in the following scenarios: 

- A protocol run has been cancelled (yellow warning banner)
- A protocol run has failed (red error banner)
- A protocol run has completed (green success banner)

When an uncleared + inactive protocol is loaded onto the robot, and the user tries to start a new protocol run, we should: 

- Allow them to confirm in one click via a modal dialog that they want to clear the previous and proceed to the new run. (Instead have having to navigate back to the previous run, clear it, and then start the next protocol run.)

# Acceptance Criteria

- [ ] The clear protocol button appears in all of the correct protocol run scenarios listed above
- [ ] The clear protocol banner appears in all of the correct protocol run scenarios listed above
- [ ] The clear protocol and proceed with run modal dialog allows users to clear and proceed with one click

### PR DESCRIPTION
**Pull request by brenthagen**
# Overview

this modal completes the "clear protocol" updates that allow users to clear a protocol from a "busy" robot. the modal appears from the "choose robot" slideout.

Screen Shot 2022-04-20 at 7 54 20 PM

closes #9775

# Changelog

 - Adds "robot is busy" modal to choose protocol slideout

# Review requests

confirm modal works as expected when selecting a robot with a protocol already loaded (a current run exists for that robot)

# Risk assessment

low

### CODE REVIEW

File: app/src/organisms/ChooseRobotSlideout/__tests__/ChooseRobotSlideout.test.tsx [hunk 1]
```diff
@@ -117,4 +157,40 @@ describe('ChooseRobotSlideout', () => {
...
+    })
+    const proceedButton = getByRole('button', { name: 'Proceed to setup' })
+    expect(proceedButton).toBeDisabled()
+    const mockRobot = getByText('opentrons-robot-name')
+    mockRobot.click()
+    expect(proceedButton).not.toBeDisabled()
+    proceedButton.click()
+    expect(mockCreateRun).toBeCalled()
+  })
+  it('launches the robot is busy modal when the selected robot is busy', async () => {
```
Comment (koji) [inline app/src/organisms/ChooseRobotSlideout/__tests__/ChooseRobotSlideout.test.tsx]: Looks like there is no `await`. Do you need `async`?

**Review by koji — APPROVED** [review]
Looks good to me.

I noticed one difference between `app/src/organisms/ChooseRobotSlideout/index.tsx` and design which is Line 155.
 
If you update Line 155  from `shared: refresh` to `shared:refresh_list`, that will be great.

**Review by jerader — APPROVED** [review]
LGTM 🦕

File: app/src/organisms/ChooseRobotSlideout/__tests__/ChooseRobotSlideout.test.tsx (hunk 1 above)
Comment (brenthagen) [reply app/src/organisms/ChooseRobotSlideout/__tests__/ChooseRobotSlideout.test.tsx]: no - i was tinkering with `waitFor` and forgot to remove it. thanks

**Review by smb2268 — APPROVED** [review]
Looks great!

**Pull request merged**